cmd/salt-updater/service.go

```diff
@@ -50,25 +59,104 @@ func genIntrospectable(v interface{}) introspect.Introspectable {
 	return introspect.NewIntrospectable(node)
 }
 
-//IsRunning will return true if a salt update is currently running
+// IsRunning will return true if a salt update is currently running
 func (s service) IsRunning() (bool, *dbus.Error) {
 	return s.saltUpdater.state.RunningUpdate, nil
 }
 
-//RunUpdate will start a salt update if one is not already running
+// RunUpdate will start a salt update if one is not already running
 func (s service) RunUpdate() *dbus.Error {
-	s.saltUpdater.runSaltCall([]string{"state.apply", "--state-output=mixed", "--output-diff"}, true)
+	updateAvailable, updateTime, err := UpdateExists()
+	if err != nil {
+		log.Printf("Error checking if update exists %v will run salt state", err)
+	}
+	//if we have an error lets just run salt update
+	if err == nil && !updateAvailable {
+		log.Println("No update available")
+		return nil
+	}
+	s.saltUpdater.runSaltCall([]string{"state.apply", "--state-output=mixed", "--output-diff"}, true, updateTime)
 	return nil
 }
 
-//RunPing will send a test ping to the salt server
+// updateExists checks if there has been any git updates since the last update time for this minions nodegroup
+// uses github api to view last commit to the repo
+func UpdateExists() (bool, time.Time, error) {
+	nodegroupOut, err := ioutil.ReadFile("/etc/cacophony/salt-nodegroup")
+	nodeGroup := string(nodegroupOut)
+	nodeGroup = strings.TrimSuffix(nodeGroup, "\n")
+	hyphenIndex := strings.Index(nodeGroup, "-")
+	if hyphenIndex != -1 {
+		nodeGroup = nodeGroup[hyphenIndex+1:]
```

Comment:
Are we wanting this to work with the older cameras? Because the older cameras have nodegroup like `dev-pis` instead of `tc2-dev`.
Might be better just to map each nodegroup to a branch manually?